### PR TITLE
Add the E suffix to all data types representing effects

### PR DESCRIPTION
**Pull request by arybczak**
Testing the waters for a variant proposed in #23 that advocates adding `E` to the end of all effect data types.

To be fair, I don't really like it, looks messy to me. Thoughts?

### CODE REVIEW

**Comment by mmhat** [comment]
TLDR; I'm ok without the 'E'; We can work around the issue with clever naming and if that's not possible I have no problems with clashing names.

Long answer: As you may have noted I changed my opinion on this topic while discussing #23 :
It since it does only affect `AsyncE` (IOE being the other one; But somehow I don't feel like we should change that) which we might generalize to `Conc` anyway I am fine if we do not an 'E' suffix everywhere.
Note that I chose a different path when I was working on the `typed-process`-based `Process` effect: Instead of using a suffix I went with qualified imports. `polysemy` went with that (e.g. [here](https://hackage.haskell.org/package/polysemy-1.6.0.0/docs/Polysemy-Async.html)) and it seems to work for them. Then the user has to solve the ambiguity problem which IMHO is ok since the needs might be different there. Of course the downside is that we cannot provide a "full" drop-in solution since we cannot reexport all names of the module we are "lifting".
The only thing which annoyed me while implementing the `Process` effect was that `runProcess` was taken by `typed-process`, too.

**Comment by arybczak** [comment]
> polysemy went with that (e.g. here) and it seems to work for them

I don't think `polysemy` is a good template here. For one, the definition of `Async` looks extremely confusing in the docs because `Async` refers to both `Polysemy.Async.Async` and `Control.Concurrent.Async.Async`, but it's hidden unless you hover over links. Also, the effect is sufficiently weird (the `Maybe` in `Async`, no `withAsync` to name a few) that I presume hardly anyone uses it.

> Then the user has to solve the ambiguity problem which IMHO is ok since the needs might be different there

That's not a good approach. Instead of solving the problem once in the library, you just force everyone to solve it individually in their code.

These primitive effects should be a drop-in replacement to the originals, i.e. you import `Effectful.Process.Typed` instead of `System.Process.Typed` and that's it. Otherwise it's just going to be annoying.

**Comment by arybczak** [comment]
> I'm ok without the 'E'; We can work around the issue with clever naming

Yes. I think the rule "Use the suffix `E` only if there are name clashes" is easy to apply and will solve all potential name slashes, because the convention of sticking `E` at the end of type names is not taken by any library.

**Comment by mmhat** [comment]
Yeah, I admit, looking at the docs of `polysemy` again this is indeed confusing... What do you think about extending the `Async` effect to `Conc`? At least for the effects in this repository this would be a "solution by evasion" since we don't have anything left that clashes.

As far as I am concerned that is all what I had in mind when I open #23; I do not expect naming to be consistent across the whole ecosystem.

**Comment by L-as** [comment]
I think it's better to do this universally for consistency.

**Comment by ocharles** [comment]
Personally I think the `E` suffix is the wrong way to go. Only a few effect types have this conflict, but effect types show up frequently in the context of type signatures for things that use events. Given this, I think you want to optimize for effect name readability.

**Comment by arybczak** [comment]
> What do you think about extending the Async effect to Conc? Then we don't have anything left that clashes.

`Conc` collides with the `Conc` data type :smile: `Concurrent` should be fine though.

> Personally I think the E suffix is the wrong way to go. Only a few effect types have this conflict, but effect types show up frequently in the context of type signatures for things that use events. Given this, I think you want to optimize for effect name readability.

Indeed, I think occasional collisions are fine. I'll close this PR for now.
